beamer/contracts.py

```diff
@@ -51,6 +52,36 @@ def prepare_deployment_infos(
     return infos
 
 
+class ABIManager:
+    _CacheEntry = namedtuple("_CacheEntry", ("abi", "bytecode"))
+
+    def __init__(self, abi_dir: Path):
+        self.abi_dir = abi_dir
+        self._cache: dict[str, ABIManager._CacheEntry] = {}
+
+    def get_abi(self, name: str) -> str:
+        entry = self._cache.get(name)
+        if entry is None:
+            entry = self._load_entry(name)
+            self._cache[name] = entry
+        return entry.abi
+
+    def get_bytecode(self, name: str) -> str:
```

Comment:
this seems like a duplicate to me, only difference is just the attribute namely abi or bytecode

Reply:
Very little code is duplicated here, we could have a separate helper function, but I think it would end up pretty much the same code size, if not slightly bigger.